### PR TITLE
DATASET: RSPB nature reserves

### LINKED ISSUE #131: RSPB nature reserves (and wildlife centres?) data
Could potentially scrape from here? https://www.rspb.org.uk/reserves-and-events/reserves-a-z/

TCC mentioned "wildlife centres", but I’m not sure what they are.

### PR DESCRIPTION
**Pull request by alexander-griffen**
Fixes #131 
Note: Dataset csv in the shared drive - made in QGIS by finding the centroids of each reserve, then reprojecting the data into a projection that lines up the points as lat/long coordinates

### CODE REVIEW

File: hub/management/commands/import_rspb_nature_reserves.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        },
+    }
+
+    def add_data_sets(self):
+        data_set, created = DataSet.objects.update_or_create(
+            name="rspb_nature_reserves",
+            defaults={
+                "label": "RSPB Reserves",
+                "data_type": "integer",
+                "category": "place",
```
Comment (struan) [inline hub/management/commands/import_rspb_nature_reserves.py:34]: I wonder if this makes more sense as a movement thing, and also have a second data set with the names of the reserves a la national trust?

**Review by struan — CHANGES_REQUESTED** [review]
one big question, and a few small improvements/nit picks.

Also, one fixup for linting.

File: hub/management/commands/import_rspb_nature_reserves.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                "label": "RSPB Reserves",
+                "description": "Number of RSPB reserves by constituency.",
+                "source_label": "Data from the RSPB.",
+                "release_date": "September 2023",
+                "source": "https://www.rspb.org.uk",
+                "source_type": "csv",
+                "table": "areadata",
+                "default_value": {},
+                "data_url": "",
+                "is_filterable": False,
```
[the remark this answers is not in the document]
Comment (struan) [reply hub/management/commands/import_rspb_nature_reserves.py:44]: one other thing is this should have the unit_type and unit_distribution attributes :)

File: hub/management/commands/import_rspb_nature_reserves.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        for index, row in tqdm(df.iterrows(), disable=self._quiet):
+            lat = row["Latitude"]
+            lon = row["Longitude"]
+
+            self.process_lat_long(lat=lat, lon=lon, row_name=index)
+
+            if index > 0 and index % 50 == 0:
+                sleep(10)
+
+    def add_arguments(self, parser):
```
[the remark this answers is not in the document]
Comment (alexander-griffen) [reply hub/management/commands/import_rspb_nature_reserves.py:85]: Django complains if I remove the `handle()` method, so I have kept that, and removed the other lines of code from this point on

**Comment by struan** [comment]
One small question here is that looking at the centroids csv I notice that some of the entries in there are "Non-Publicised Reserve". I'm not sure what that means here and if it means we should not be putting them on the site? Or at least not listing the names?

**Review by struan — APPROVED** [review]
Module one question in the comments this looks good.

**Pull request merged**